libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -604,18 +604,18 @@ namespace pika::mpi::experimental {
                         debug(
                             str<>("CB invoke"), ptr(mpi_data_.callbacks_[index].request_), status));
 
-                    // Invoke callback (PIKA_MOVE doesn't compile here)
-                    PIKA_INVOKE(std::move(mpi_data_.callbacks_[index].cb_), status);
-
-                    pika::threads::detail::decrement_global_activity_count();
-
                     // Remove the request from our vector to prevent retesting
                     mpi_data_.requests_[index] = MPI_REQUEST_NULL;
 
                     // decrement before invoking callback to avoid race
                     // if invoked code checks in_flight value
                     --mpi_data_.all_in_flight_;
                     --mpi_data_.active_requests_size_;
+
+                    // Invoke callback (PIKA_MOVE doesn't compile here)
+                    PIKA_INVOKE(std::move(mpi_data_.callbacks_[index].cb_), status);
+
+                    pika::threads::detail::decrement_global_activity_count();
```

Comment:
The multithreaded poll also needs this change (in a couple of places)

Reply:
To clarify, the multithreaded polling is already handling the counts this way as far as I can tell. That's why you don't see any changes to it here.

Reply:
ok. ignore my comment. I misread the code.